ogr/ogrsf_frmts/hana/ogrhanadatasource.cpp

```diff
@@ -1305,6 +1300,72 @@ void OGRHanaDataSource::InitializeLayers(
     }
 }
 
+/************************************************************************/
+/*                          LaunderName()                               */
+/************************************************************************/
+
+std::pair<OGRErr, CPLString> OGRHanaDataSource::LaunderName(const char* name)
+{
+    CPLAssert(name != nullptr);
+
+    if (!CPLIsUTF8(name, -1))
+    {
+       CPLError(
+            CE_Failure, CPLE_AppDefined,
+            "%s is not a valid UTF-8 string.", name);
+        return {OGRERR_FAILURE, ""};
+    }
+
+    auto getUTF8SequenceLength = [](char c)
+    {
+        if ((c & 0x80) == 0x00)
+            return 1;
+        if ((c & 0xE0) == 0xC0)
+            return 2;
+        if ((c & 0xF0) == 0xE0)
+            return 3;
+        if ((c & 0xF8) == 0xF0)
+            return 4;
+
+        throw std::runtime_error("Invalid UTF-8 sequence");
```

Comment:
are you sure that all calling code paths catch that exception ?

Reply:
My assumption was that we can rely on a check in line 1311.
If this check is reliable than we should never reach line 1330.

Reply:
> My assumption was that we can rely on a check in line 1311.

ah right.